spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMergeOnReadDelete.java

```diff
@@ -74,14 +75,19 @@ public void testCommitUnknownException() {
 
     // write unpartitioned files
     append(tableName, "{ \"id\": 1, \"dep\": \"hr\", \"category\": \"c1\"}");
+    createBranch();
```

Comment:
given that we do not allow auto-creation of branches during write, all the tests are updated in such a way that the first write is to the table, and then create branch commit is called to make the branch from main.